engine/consensus/compliance/engine_test.go

```diff
@@ -14,74 +16,88 @@ import (
 	"github.com/onflow/flow-go/engine"
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/model/messages"
+	"github.com/onflow/flow-go/module/irrecoverable"
 	modulemock "github.com/onflow/flow-go/module/mock"
+	"github.com/onflow/flow-go/module/util"
 	"github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/network/mocknetwork"
 	"github.com/onflow/flow-go/utils/unittest"
 )
 
 func TestComplianceEngine(t *testing.T) {
-	suite.Run(t, new(ComplianceSuite))
+	suite.Run(t, new(EngineSuite))
 }
 
-type ComplianceSuite struct {
-	ComplianceCoreSuite
+// EngineSuite tests the compliance engine.
+type EngineSuite struct {
+	CommonSuite
 
+	ctx    irrecoverable.SignalerContext
+	cancel context.CancelFunc
+	errs   <-chan error
```

Comment:
Hm, this seems a bit dangerous, this channel is not read while test is running, should we spin a up a separate goroutine to read from this channel to provide a receiver?

Reply:
Using `unittest.FailOnIrrecoverableError` for instance

Reply:
👍 Good point, adding